src/realm/object-store/shared_realm.cpp

```diff
@@ -260,6 +261,24 @@ void Realm::read_schema_from_group_if_needed()
     notify_schema_changed();
 }
 
+void Realm::load_cached_schema_if_needed(uint64_t current_tr_version, Schema& schema)
```

Comment:
@tgoyne I implemented this method, following your explanation of the problem we had for the frozen realm. In order to load the latest cached schema based on what the coordinator just saw.

Please tell me, if there is another way for doing this or what I am doing, it is not right.  

The basic idea, is to compare the current transaction version vs whatever the coordinator has seen. 
If the coordinator saw a transaction version that is greater than the current one, and we are in additive mode, I am basically loading the cached schema.